SCMS.Services.Tests.Unit/Services/Processings/StudentGuardians/StudentGuardianProcessingTests.Logic.Verify.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -----------------------------------------------------------------------
+// Copyright (c) Signature Chess Club & MumsWhoCode. All rights reserved.
+// -----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Moq;
+using SCMS.Services.Api.Models.Foundations.StudentGuardians;
+using Xunit;
+
+namespace SCMS.Services.Tests.Unit.Services.Processings.StudentGuardians
+{
+    public partial class StudentGuardianProcessingTests
+    {
+        [Fact]
+        public void ShouldVerifyNoPrimaryStudentGuardianExists()
+        {
+            // given
+            Guid inputStudentId = Guid.NewGuid();
+            Guid inputGuardianId = Guid.NewGuid();
+            StudentGuardian expectedStudentGuardian = null;
+
+            IQueryable<StudentGuardian> randomStudentGuardians =
+                CreateRandomStudentGuardians();
+
+            IQueryable<StudentGuardian> storageStudentGuardians =
```

Comment:
How do you know it's storage? We are far away from the shore now, all we know that these guardians are retrieved.
We might think about calling everything broker-neighboring or not to be just retrieved across the board - let's think about that.